include/nbl/ext/ImGui/ImGui.h

```diff
@@ -1,76 +1,133 @@
 #ifndef _NBL_EXT_IMGUI_UI_H_
 #define _NBL_EXT_IMGUI_UI_H_
 
+#include "nbl/video/declarations.h"
+#include "nbl/asset/IAssetManager.h"
+
 namespace nbl::ext::imgui
 {
 class UI final : public core::IReferenceCounted
 {
 	public:
-		UI(core::smart_refctd_ptr<video::ILogicalDevice> device, int maxFramesInFlight, video::IGPURenderpass* renderpass, video::IGPUPipelineCache* pipelineCache, core::smart_refctd_ptr<ui::IWindow> window);
+		struct MDI
+		{
+			using COMPOSE_T = nbl::video::StreamingTransientDataBufferST<nbl::core::allocator<uint8_t>>;								//! composes memory available for the general purpose allocator to suballocate memory ranges								
+			using SUBALLOCATOR_TRAITS_T = nbl::core::address_allocator_traits<nbl::core::LinearAddressAllocatorST<uint32_t>>;			//! traits for MDI buffer suballocator - fills the data given the mdi allocator memory request
+
+			enum E_BUFFER_CONTENT : uint8_t
+			{
+				EBC_DRAW_INDIRECT_STRUCTURES,
+				EBC_ELEMENT_STRUCTURES,
+				EBC_INDEX_BUFFERS,
+				EBC_VERTEX_BUFFERS,
+
+				EBC_COUNT,
+			};
+
+			nbl::core::smart_refctd_ptr<typename COMPOSE_T> buffer;							//! streaming mdi buffer
+
+			static constexpr auto MDI_BUFFER_REQUIRED_ALLOCATE_FLAGS = nbl::core::bitflag<nbl::video::IDeviceMemoryAllocation::E_MEMORY_ALLOCATE_FLAGS>(nbl::video::IDeviceMemoryAllocation::EMAF_DEVICE_ADDRESS_BIT); //! required flags
+			static constexpr auto MDI_BUFFER_REQUIRED_USAGE_FLAGS = nbl::core::bitflag(nbl::asset::IBuffer::EUF_INDIRECT_BUFFER_BIT) | nbl::asset::IBuffer::EUF_INDEX_BUFFER_BIT | nbl::asset::IBuffer::EUF_VERTEX_BUFFER_BIT | nbl::asset::IBuffer::EUF_SHADER_DEVICE_ADDRESS_BIT; //! required flags
```

Comment:
avoid duplicate prefixes, when you type this out you end up with `MDI::MDI_`

Also I'd use CamelCase starting with a major letter for constexr and leave ALL caps with _ for macros